reddit_liveupdate/controllers.py

```diff
@@ -451,18 +452,62 @@ def GET_about(self):
     )
     def GET_discussions(self, num, after, reverse, count):
         """Get a list of reddit submissions linking to this thread."""
-        builder = url_links_builder(
-            url="/live/" + c.liveupdate_event._id,
-            num=num,
-            after=after,
-            reverse=reverse,
-            count=count,
-        )
+
+        show_hidden = c.liveupdate_permissions.allow("discussions")
+        builder = get_discussions(
+            c.liveupdate_event, limit=50, show_hidden=show_hidden)
         listing = LinkListing(builder).listing()
+        listing.render_class = pages.LiveUpdateDiscussionsListing
+        listing.submit_url = pages.make_submit_url(c.liveupdate_event)
         return pages.LiveUpdateEventPage(
             content=listing,
         ).render()
 
+    @require_oauth2_scope("livemanage")
+    @validatedForm(
+        VLiveUpdateContributorWithPermission("discussions"),
+        VModhash(),
+        link=VLink("link"),
+    )
+    @api_doc(
+        section=api_section.live,
+    )
+    def POST_hide_discussion(self, form, jquery, link):
+        """Hide a linked comment thread from the discussions sidebar and listing.
+
+        Requires the `discussions` permission for this thread.
+
+        See also: [/api/live/*thread*/unhide_discussion](#POST_api_live_{thread}_unhide_discussion).
+
+        """
+        url = pages.make_event_url(c.liveupdate_event._id)
+        if link.url != url:
+            c.errors.add(errors.LIVEUPDATE_LINK_IS_NOT_DISCUSSION)
+            form.set_error(errors.LIVEUPDATE_LINK_IS_NOT_DISCUSSION, None)
+            return
+
+        c.liveupdate_event.hide_discussion(link)
+        _broadcast(type="hide_discussion", payload={"link_id": link._id36})
+
+    @require_oauth2_scope("livemanage")
+    @validatedForm(
+        VLiveUpdateContributorWithPermission("discussions"),
+        VModhash(),
+        link=VLink("link"),
+    )
+    @api_doc(
+        section=api_section.live,
+    )
+    def POST_unhide_discussion(self, form, jquery, link):
+        """Unhide a linked comment thread from the discussions sidebar and listing..
+
+        Requires the `discussions` permission for this thread.
+
+        See also: [/api/live/*thread*/hide_discussion](#POST_api_live_{thread}_hide_discussion).
+
+        """
```

Comment:
I see that it's a noop if the link isn't a hidden discussion on the event, but you could do a similar check on the link's url like you do on the hide action above.